src/app/common/services/project-service.coffee

```diff
@@ -29,10 +29,16 @@ angular.module("doubtfire.common.services.projects", [])
     ]
 
     for t in times
-      diff = laterTime.diff(earlyTime, t)
-      if diff > 1
+      #exactDiff is floating point
+      exactDiff = laterTime.diff(earlyTime, t, true).toFixed(2)
+      diff = Math.floor(exactDiff)
+      # if days are more than 14 then show in week
+      if(exactDiff > 2 && t == "weeks")
         return "#{diff} #{t.charAt(0).toUpperCase() + t.substr(1)}"
-      else if diff == 1
+      # Always show in days, Hours, Minutes and Seconds.
+      else if diff > 1 && t != "weeks"
```

Comment:
Why diff and not exactDiff here?

Reply:
and below

Reply:
If we call exactDiff then it will create some problems. Like, let's say if the deadline was missed by 1 hour 20 minutes. So if we use exact diff then the condition "exactDiff > 1" will be true but the message it shows is "#{diff} #{t.charAt(0).toUpperCase() + t.substr(1)}" which will show as "1 hours" and not "1 hour". And another reason not to use exactDiff is that we are anyways showing the floor value and not the exact value.

ExactDiff was just for the week part and to handle the condition if exactly 14 days were left.